core/src/main/java/org/apache/brooklyn/core/mgmt/classloading/JavaBrooklynClassLoadingContext.java

```diff
@@ -82,6 +87,10 @@ private ClassLoader getClassLoader() {
         try {
             className = DeserializingClassRenamesProvider.findMappedName(className);
             return (Maybe) Maybe.of(getClassLoader().loadClass(className));
+        } catch (NoClassDefFoundError e) {
+            String msg = "Invalid linkage in (transitive dependencies of) class "+className+": "+e.toString();
+            LOG.debug(msg);
```

Comment:
As you said in the comment I prefer to be warning.
Users are not supposed to use invalid classes in their apps.